skillserver/skillserver.go

```diff
@@ -58,12 +59,38 @@ func SetRootPrefix(prefix string) {
 	RootPrefix = prefix
 }
 
+// RunSSL takes in a map of application, server port, certificate and key file names, and
+// tries to start a TLS server which alexa can directly pass commands to.
+// It returns back an error if the server couldn't be started. Or else the method blocks
+// at ListenAndServeTLS line
+// IF the server starts succcessfully and there are connection errors afterwards, they are
+// logged to the stdout and no error is returned
+// For generating a testing cert and key, read the following:
+// https://developer.amazon.com/docs/custom-skills/configure-web-service-self-signed-certificate.html
 func RunSSL(apps map[string]interface{}, port, cert, key string) error {
-	router := mux.NewRouter()
-	Init(apps, router)
-
-	err := http.ListenAndServeTLS(port, cert, key, router)
-	return err
+  router := mux.NewRouter()
+  Init(apps, router)
+
+  // This is very limited TLS configuration which is required to connect alexa to our webservice.
+  // The curve preferences are used by ECDSA/ECDHE algorithms for figuring out the mathing algorithm
+  // from alexa side starting from the strongest to the weakest
+  cfg := &tls.Config{
+    MinVersion:               tls.VersionTLS12,
+    CurvePreferences:         []tls.CurveID{tls.CurveP521, tls.CurveP384, tls.CurveP256},
+    PreferServerCipherSuites: true,
+    CipherSuites: []uint16{
+      //If the connection throws errors related to crypt algorithm mismatch between server and client,
```

Comment:
- nit 1: there should be a space after `//`
- nit 2: missing a period at the end of the comment.